src/Sylius/Behat/Context/Api/Admin/ManagingTaxRatesContext.php

```diff
@@ -0,0 +1,526 @@
+        }
+        $taxRateToDelete[] = $taxRate->getCode();
+        $this->sharedStorage->set('tax_rate_to_delete', $taxRateToDelete);
+    }
+
+    /**
+     * @When I delete them
+     */
+    public function iDeleteThem(): void
+    {
+        foreach ($this->sharedStorage->get('tax_rate_to_delete') as $id) {
+            $this->client->delete(Resources::TAX_RATES, (string) $id)->getContent();
+        }
+    }
+
+    /**
+     * @Then I should be notified that they have been successfully deleted
+     */
+    public function iShouldBeNotifiedThatTheyHaveBeenSuccessfullyDeleted(): void
+    {
+        Assert::true(
+            $this->responseChecker->isDeletionSuccessful($this->client->getLastResponse()),
+            'Tax rate could not be deleted'
+        );
+    }
+
+    /**
+     * @Then I should see a single tax rate in the list
+     */
+    public function iShouldSeeASingleTaxRateInTheList(): void
+    {
+        Assert::same($this->responseChecker->countCollectionItems($this->client->index(Resources::TAX_RATES)), 1);
+    }
+
+    /**
+     * @Then I should be notified that tax rate with this code already exists
+     */
+    public function iShouldBeNotifiedThatTaxRateWithThisCodeAlreadyExists(): void
+    {
+        $response = $this->client->getLastResponse();
+        Assert::false(
+            $this->responseChecker->isCreationSuccessful($response),
+            'Tax rate has been created successfully, but it should not'
+        );
+        Assert::same(
+            $this->responseChecker->getError($response),
+            'code: The tax rate with given code already exists.'
+        );
+    }
+
+    /**
+     * @Then there should still be only one tax rate with code :code
+     */
+    public function thereShouldStillBeOnlyOneTaxRateWithCode(string $code): void
+    {
+        Assert::count(
+            $this->responseChecker->getCollectionItemsWithValue($this->client->index(Resources::TAX_RATES), 'code', $code),
+            1,
+            sprintf('There is more than one tax rate with code %s', $code)
+        );
+    }
+
+    /**
+     * @Then I should be notified that :element is required
+     */
+    public function iShouldBeNotifiedThatCodeIsRequired(string $element): void
+    {
+        Assert::contains(
+            $this->responseChecker->getError($this->client->getLastResponse()),
+            sprintf('%s: Please enter tax rate %s.', $element, $element)
+        );
+    }
+
+    /**
+     * @Then tax rate with :element :code should not be added
+     */
+    public function taxRateWithCodeShouldNotBeAdded(string $element, string $code): void
+    {
+        Assert::false($this->isItemOnIndex($element, $code), sprintf('Tax rate with %s %s exist', $element, $code));
+    }
+
+    private function isItemOnIndex(string $property, string $value): bool
+    {
+        return $this->responseChecker->hasItemWithValue($this->client->index(Resources::TAX_RATES), $property, $value);
+    }
```

Comment:
This private method should be moved below public ones

Reply:
And I'm not sure if it is needed as it is used only once and it consists of one simple line 🤔